types/redux-form/index.d.ts

```diff
@@ -10,37 +10,54 @@ import {
   ReactElement
 } from "react";
 
-export type FieldValue = any;
-
 export type FieldType = "Field" | "FieldArray";
 
-export interface DataShape {
-    [fieldName: string]: FieldValue
+export interface ErrorOther<T = void> {
+    _error?: T;
 }
 
-export type FormErrors<FormData extends DataShape> = {
-    [P in keyof FormData]?: ReactElement<any> | string | { _error?: string };
+export type FormErrors<FormData = {}, T = void> = {
+    [P in keyof FormData]?: ReactElement<any> | string | ErrorOther<T>;
 };
 
-export type FormWarnings<FormData extends DataShape> = {
-    [P in keyof FormData]?: ReactElement<any> | string | { _warning?: string };
+export interface WarningOther<T = void> {
+    _warning?: T;
+}
+
+export type FormWarnings<FormData = {}, T = void> = {
+    [P in keyof FormData]?: ReactElement<any> | string | WarningOther<T>;
 };
 
+export interface RegisteredFieldState {
+    name: string;
+    type: FieldType;
+}
+
+
+//export type Obj<T> = { [k: string]: T };
+//export type ObjectHasKey<O extends {}, K extends string> = ({[K in keyof O]: '1' } & Obj<'0'>)[K];
+export type Diff<T extends string, U extends string> = ({ [P in T]: P } & { [P in U]: never } & { [x: string]: never })[T];
+export type Omit<T, K extends keyof T> = Pick<T, Diff<keyof T, K>>;
+//export type Overwrite<K, T> = {[P in keyof T | keyof K]: { 1: T[P], 0: K[P] }[ObjectHasKey<T, P>]};
```

Comment:
Those were copy-pasted from official docs. I can't copy-paste them from the docs but don't really see the point in doing that.